### PR TITLE
Implement iptable rules capabilities

### PR DESCRIPTION
**Pull request by kratos7**
Signed-off-by: Abdullah Shah shah.abdullah.syed@gmail.com

### CODE REVIEW

File: src/main/python/hydra/lib/childmgr.py [hunk 1]
```diff
@@ -91,4 +91,7 @@ def terminate_process_and_children(self, name):
         children = parent_proc.children(recursive=True)
         for proc in children:
             l.debug(proc)
-            proc.send_signal(signal.SIGKILL)
+            try:
```
Comment (sushilks) [inline src/main/python/hydra/lib/childmgr.py:94]: was this the problem with pypy run's ?

File: src/main/python/hydra/rmqtest/runtest.py [hunk 2]
```diff
@@ -49,6 +49,7 @@ def __init__(self, options, runtest=True, mock=False):
             self.stop_appserver()
 
     def rerun_test(self, options):
+        l.info("THIS IS A TEST")
```
Comment (sushilks) [inline src/main/python/hydra/rmqtest/runtest.py]: leftover ?

File: src/main/python/hydra/kraken/kraken.py [hunk 3]
```diff
@@ -21,14 +22,18 @@
...
         RunTestBase.__init__(self, 'kraken', self.options, self.config, startappserver=runtest, mock=mock)
 
     def release_the_kraken(self):
         """
         soon...
         """
+        ip_to_block = "200.200.200.200"
+        ip_to_unblock = "200.200.200.200"
+        self.k_api.block_ip_on_node(ip_to_block)
+        self.k_api.unblock_ip_on_node(ip_to_unblock)
```
Comment (sushilks) [inline src/main/python/hydra/kraken/kraken.py]: it's not really testing the block /unblock right ?
probably can do netcat or some similar command to test block/no-block.

File: src/main/python/hydra/kraken/kraken_api.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+class KrakenApi(object):
+    def __init__(self):
+        l.info("Kraken_api initiated")
+
+    def block_ip_port_on_node(self, ip_to_block, port, chain="INPUT", protocol="tcp", host_ip="", user=""):
+        """
+        Blocks all incoming communication from an ip on a host (local or remote)
+        @args:
+        ip_to_block:     IP to block
```
Comment (sushilks) [inline src/main/python/hydra/kraken/kraken_api.py:19]: we may need to updated it with wildcard support for src ip / port.

Comment (kratos7) [inline src/main/python/hydra/kraken/kraken_api.py:19]: yes this is just a POC sort of. We will enhance it as we gain more visibility.

**Pull request merged**